pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -280,7 +280,15 @@ func ValidateACMEChallengeSolverDNS01(p *cmacme.ACMEChallengeSolverDNS01, fldPat
 			el = append(el, field.Forbidden(fldPath.Child("cloudflare"), "may not specify more than one provider type"))
 		} else {
 			numProviders++
-			el = append(el, ValidateSecretKeySelector(&p.Cloudflare.APIKey, fldPath.Child("cloudflare", "apiKeySecretRef"))...)
+			if p.Cloudflare.APIKey != nil {
+				el = append(el, ValidateSecretKeySelector(p.Cloudflare.APIKey, fldPath.Child("cloudflare", "apiKeySecretRef"))...)
+			}
+			if p.Cloudflare.APIToken != nil {
+				el = append(el, ValidateSecretKeySelector(p.Cloudflare.APIToken, fldPath.Child("cloudflare", "apiTokenSecretRef"))...)
+			}
```

Comment:
We should also validate here that only one of APIToken or APIKey is non-nil - it isn't valid for *both* to be set

Reply:
(+ corresponding tests)

Reply:
I've pushed a commit up to fix this 😄